MantidQt/CustomInterfaces/src/Indirect/AbsorptionCorrections.cpp

```diff
@@ -301,29 +298,44 @@ void AbsorptionCorrections::algorithmComplete(bool error) {
                                          m_uiForm.spCanShift->value()));
     shiftLog->execute();
   }
-  // Plot output
-
-  bool plot = m_uiForm.ckPlot->isChecked();
-  if (plot) {
-    QStringList plotData = {QString::fromStdString(m_pythonExportWsName),
-                            m_uiForm.dsSampleInput->getCurrentDataName()};
-    auto outputFactorsWsName =
-        m_absCorAlgo->getPropertyValue("CorrectionsWorkspace");
-    if (m_uiForm.ckKeepFactors->isChecked()) {
-      QStringList plotCorr = {QString::fromStdString(outputFactorsWsName) +
-                              "_ass"};
-      if (m_uiForm.ckUseCanCorrections->isChecked()) {
-        plotCorr.push_back(QString::fromStdString(outputFactorsWsName) +
-                           "_acc");
-        QString shiftedWs = QString::fromStdString(
-            m_absCorAlgo->getPropertyValue("CanWorkspace"));
-        plotData.push_back(shiftedWs);
-      }
-      plotSpectrum(plotCorr, 0);
+  // Enable plot and save
+  m_uiForm.pbPlot->setEnabled(true);
+  m_uiForm.pbSave->setEnabled(true);
+}
+/**
+ * Handle saving of workspace
+ */
+void AbsorptionCorrections::saveClicked() {
+
+  checkADSForPlotSaveWorkspace(m_pythonExportWsName, false);
+  addSaveWorkspace(QString::fromStdString(m_pythonExportWsName));
+  bool keepCorrectionFactors = m_uiForm.ckKeepFactors->isChecked();
+  checkADSForPlotSaveWorkspace(m_outputFactorsWsName.toStdString(), false);
+  if (keepCorrectionFactors)
+    addSaveWorkspace(m_outputFactorsWsName);
```

Comment:
Just a very minor comment: if `keepCorrectionFactors == False` I assume we don't even need to check if the corresponding workspace is in the ADS, so we could avoid this check by moving line 313 inside the `if` statment.